src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.model.transaction;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.transaction.stub.Category;
+import seedu.address.model.transaction.stub.Description;
+
+
+
+/**
+ * Represents a Transaction in a TransactionList.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Transaction {
+
+    private final Date date;
+    private final Amount amount;
+    private final Description description;
+    private final Set<Category> categories = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Transaction(Date date, Amount amount, Description description, Set<Category> categories) {
+        requireAllNonNull(date, amount, description, categories);
+        this.date = date;
+        this.amount = amount;
+        this.description = description;
+        this.categories.addAll(categories);
+    }
```

Comment:
Transaction needs direction, i.e. a type of either credit or debit. This assumes that amount will always be positive.